kvbc/include/db_interfaces.h

```diff
@@ -96,6 +96,10 @@ class IBlocksDeleter {
   // Throws on errors or if until <= genesis .
   virtual BlockId deleteBlocksUntil(BlockId until) = 0;
 
+  // This method should get the last block ID in the system and deletes it.
+  // This method will increment the genesis block id.
+  virtual void deleteLastReachableBlock() = 0;
```

Comment:
Actually, it deletes the newest block that was added i.e. has the highest id, and deletes it.
It doesn't impact the genesis (which is the oldest).